src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs

```diff
@@ -53,19 +55,50 @@ internal sealed class ClassEvaluator : MemberEvaluator {
                 // Set bases to the class.
                 var bases = new List<IPythonType>();
                 foreach (var a in _classDef.Bases.Where(a => string.IsNullOrEmpty(a.Name))) {
-                    // We cheat slightly and treat base classes as annotations.
-                    var b = Eval.GetTypeFromAnnotation(a.Expression);
-                    if (b != null) {
-                        var t = b.GetPythonType();
-                        bases.Add(t);
-                        t.AddReference(Eval.GetLocationOfName(a.Expression));
+                    // can either be the name of a class declared in the current module
+                    // or a class declared in some other module
+                    var cur = a.Expression as NameExpression;
+                    var mod = a.Expression as MemberExpression;
+                    if (cur == null && mod == null) {
```

Comment:
Please use `switch` pattern matching and consider better names than `cur` and `mod` to represent either a type of the expression (name/member) or the semantics of the variable. 
```cs
switch (a.Expression) {
    case NameExpression nameExpression:
...
        break;
    case MemberExpression memberExpression:
...
        break;
    default:
...
        break;
}
```

Reply:
Done!